plugin/http/httptrace/httptrace.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package httptrace contains OpenCensus tracing integrations with net/http.
+package httptrace
+
+import (
+	"net/http"
+	"strings"
+
+	"go.opencensus.io/trace"
+	"go.opencensus.io/trace/propagation"
+)
+
+const httpHeader = `X-Cloud-Trace-Context`
+
+type transport struct {
+	base        http.RoundTripper
+	propogators []propagation.HTTPPropagator
+}
+
+// RoundTrip creates a trace.Span and inserts it into the outgoing request's headers.
+// The created span can follow a parent span, if a parent is presented in
+// the request's context.
+func (t *transport) RoundTrip(req *http.Request) (*http.Response, error) {
+	name := "Sent" + strings.Replace(req.URL.String(), req.URL.Scheme, ".", -1)
+	ctx := trace.StartSpan(req.Context(), name)
+	req = req.WithContext(ctx)
+
+	span := trace.FromContext(ctx)
+	for _, p := range t.propogators {
+		req = p.ToRequest(span.SpanContext(), req)
+	}
+
+	resp, err := t.base.RoundTrip(req)
+
+	// TODO(jbd): Add status and attributes.
+	trace.EndSpan(ctx)
+	return resp, err
+}
+
+// CancelRequest cancels an in-flight request by closing its connection.
+func (t *transport) CancelRequest(req *http.Request) {
+	type canceler interface {
+		CancelRequest(*http.Request)
+	}
+	if cr, ok := t.base.(canceler); ok {
+		cr.CancelRequest(req)
+	}
+}
+
+// NewTransport returns an http.RoundTripper that traces the outgoing requests.
+//
+// All the requests are done by the given base roundtripper. If nil is given,
+// http.DefaultTransport is used.
+//
+// Traces are propagated via the provided HTTP propagation mechanisms.
+func NewTransport(base http.RoundTripper, p ...propagation.HTTPPropagator) http.RoundTripper {
+	if base == nil {
+		base = http.DefaultTransport
+	}
+	return &transport{base: base, propogators: p}
+}
+
+// NewHandler returns a http.Handler from the given handler
+// that is aware of the incoming request's span.
+// The span can be extracted from the incoming request in handler
+// functions from incoming request's context:
+//
+//    span := trace.FromContext(r.Context())
+//
+// The span will be auto finished by the handler.
```

Comment:
Let's make this more clear
```go
The span will be automatically ended by the handler
```
if I understood the intention of the code and the above statement, properly.